website/static/js/projectSettings.js

```diff
@@ -8,77 +8,114 @@ var $osf = require('js/osfHelpers');
 var oop = require('js/oop');
 var ChangeMessageMixin = require('js/changeMessage');
 
-var NodeCategorySettings = oop.extend(
+var ProjectSettings = oop.extend(
     ChangeMessageMixin,
     {
-        constructor: function(category, categories, updateUrl, disabled) {
+        constructor: function(params) {
             this.super.constructor.call(this);
-
             var self = this;
 
-            self.disabled = disabled || false;
-            self.UPDATE_SUCCESS_MESSAGE = 'Category updated successfully';
-            self.UPDATE_ERROR_MESSAGE = 'Error updating category, please try again. If the problem persists, email ' +
+            self.decodedTitle = params.currentTitle;
+            self.decodedDescription = $osf.htmlDecode(params.currentDescription);
+            self.title = ko.observable(params.currentTitle).extend({
+                required: {
+                    params: true,
+                    message: 'Title cannot be blank.'
+                }});
+            self.description = ko.observable(self.decodedDescription);
+
+            self.disabled = params.disabled || false;
+
+            self.UPDATE_SUCCESS_MESSAGE = 'Category, title, and description updated successfully';
+            self.UPDATE_CATEGORY_ERROR_MESSAGE = 'Error updating category, please try again. If the problem persists, email ' +
                 '<a href="mailto:support@osf.io">support@osf.io</a>.';
-            self.UPDATE_ERROR_MESSAGE_RAVEN = 'Error updating Node.category';
+            self.UPDATE_DESCRIPTION_ERROR_MESSAGE = 'Error updating description, please try again. If the problem persists, email ' +
+                '<a href="mailto:support@osf.io">support@osf.io</a>.';
+            self.UPDATE_CATEGORY_ERROR_MESSAGE_RAVEN = 'Error updating Node.category';
 
-            self.INSTANTIATION_ERROR_MESSAGE = 'Trying to instantiate NodeCategorySettings view model without an update URL';
+            self.INSTANTIATION_ERROR_MESSAGE = 'Trying to instantiate ProjectSettings view model without an update URL';
 
             self.MESSAGE_SUCCESS_CLASS = 'text-success';
             self.MESSAGE_ERROR_CLASS = 'text-danger';
 
-            if (!updateUrl) {
+            if (!params.updateUrl) {
                 throw new Error(self.INSTANTIATION_ERROR_MESSAGE);
             }
 
-            self.categories = categories;
-            self.category = ko.observable(category);
-            self.updateUrl = updateUrl;
-
-            self.selectedCategory = ko.observable(category);
-            self.dirty = ko.observable(false);
-            self.selectedCategory.subscribe(function(value) {
-                if (value !== self.category()) {
-                    self.dirty(true);
-                }
-            });
+            self.categories = params.categories;
+            self.category = params.category;
+            self.updateUrl = params.updateUrl;
+            self.selectedCategory = ko.observable(params.category);
```

Comment:
What is the difference between category and selected category? Is it initial state for the cancel button?